driftbase/api/richpresence.py

```diff
@@ -0,0 +1,67 @@
+"""
+Rich Presence is player metadata which is relevant to others, such as a players online status. Rich presence
+can be queried from this extension, or you can listen to the message queue to get live updates without polling.
+"""
+
+from flask.views import MethodView
+from driftbase.flask.flaskproxy import g
+from driftbase.models.db import CorePlayer
+from marshmallow import Schema, fields
+from drift.blueprint import Blueprint
+import http.client as http_client
+from driftbase.richpresence import RichPresenceSchema, get_richpresence
+from drift.core.extensions.urlregistry import Endpoints
+from flask import url_for
+from webargs.flaskparser import abort
+
+
+bp = Blueprint("richpresence", __name__, url_prefix="/rich-presence/")
+endpoints = Endpoints()
+
+def drift_init_extension(app, **kwargs):
+    app.register_blueprint(bp)
+    endpoints.init_app(app)
+
+class RichPresenceRequestSchema(Schema):
+    name = fields.List(fields.String())
+
+class RichPresenceListArgs(Schema):
+    class Meta:
+        strict = True
+
+    player_id = fields.List(
+        fields.Integer(), metadata=dict(description="Player ID's to filter for"
+    ))
+
+@bp.route('/<int:player_id>', endpoint='entry')
+class RichPresenceAPI(MethodView):
```

Comment:
Could there be a reason for not allowing you to query for arbitrary players that you're not friends with or in a match with?

Reply:
Protected it and added tests.